### PR TITLE
Added a search function to the groups screen

### LINKED ISSUE #302: Add search to GroupsScreen
### What
A search button like on the other screens, that searches through all groups (committees, societies, boards).

### Why
I guess it's not that important but makes it easy to find a specific committee.

### Context
Create an `AllGroupsCubit` when the search delegate is opened (it doesn't need to be provided across the full app).

### PR DESCRIPTION
**Pull request by JeeVee11**
Closes #302.

### Summary
Added a search function to the groups screen.

### How to test
Steps to test the changes you made:
1. Go to the groups screen
2. Click on the search button
3. ...
4. Profit

### CODE REVIEW

File: lib/blocs/groups_cubit.dart [hunk 1]
```diff
@@ -47,3 +51,52 @@ class CommitteesCubit extends GroupsCubit {
...
+  String? _searchQuery;
+
+  /// A timer used to debounce calls to `this.load()` from `this.search()`.
+  Timer? _searchDebounceTimer;
+
+  AllGroupsCubit(ApiRepository api) : super(api, null);
+
+  @override
+  Future<void> load() async {
+    // print("AAA");
```
Comment (JAicewizard) [inline lib/blocs/groups_cubit.dart:66]: Remove this

File: lib/blocs/groups_cubit.dart [hunk 2]
```diff
@@ -47,3 +51,52 @@ class CommitteesCubit extends GroupsCubit {
...
+    emit(const LoadingState());
+    try {
+      final query = _searchQuery;
+
+      if (query != _searchQuery) return;
```
Comment (JAicewizard) [inline lib/blocs/groups_cubit.dart:71]: This check if query changed should be called after getting the response from the api

File: lib/ui/screens/groups_screen.dart [hunk 3]
```diff
@@ -1,11 +1,14 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:google_fonts/google_fonts.dart';
 import 'package:reaxit/blocs/detail_state.dart';
 import 'package:reaxit/blocs/groups_cubit.dart';
 import 'package:reaxit/models/group.dart';
 import 'package:reaxit/ui/widgets.dart';
 import 'package:collection/collection.dart';
 
+import '../../api/api_repository.dart';
```
Comment (JAicewizard) [inline lib/ui/screens/groups_screen.dart:10]: `import 'package:reaxit/api/exceptions.dart';`

File: lib/ui/screens/groups_screen.dart [hunk 4]
```diff
@@ -173,3 +193,81 @@ class GroupListScrollView extends StatelessWidget {
...
+            key: const PageStorageKey('groups-search'),
+            groups: state.result ?? [],
+          );
+        }
+      },
+    );
+  }
+
+  @override
+  Widget buildSuggestions(BuildContext context) {
```
Comment (JAicewizard) [inline lib/ui/screens/groups_screen.dart:257]: You can just call `buildResults` here

File: lib/blocs/groups_cubit.dart [hunk 5]
```diff
@@ -1,8 +1,12 @@
+import 'dart:async';
+
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:reaxit/api/api_repository.dart';
 import 'package:reaxit/api/exceptions.dart';
-import 'package:reaxit/blocs/detail_state.dart';
 import 'package:reaxit/models/group.dart';
+import 'package:reaxit/config.dart' as config;
+
+import 'detail_state.dart';
```
Comment (DeD1rk) [inline lib/blocs/groups_cubit.dart:9]: Could you make this a package import (package:reaxit/...) or perhaps just import .../models.dart)

**Review by DeD1rk — DISMISSED** [review]
Nice!

**Review by DeD1rk — APPROVED** [review]

**Pull request merged**